pallets/automation-time/src/fees.rs

```diff
@@ -194,6 +267,100 @@ mod tests {
 		})
 	}
 
+	#[test]
+	fn call_pay_checked_fees_for_with_normal_flow_and_enough_execution_fee_success() {
+		new_test_ext(0).execute_with(|| {
+			let destination = MultiLocation::new(1, X1(Parachain(PARA_ID)));
+			let alice = AccountId32::new(ALICE);
+			let mut spy = 0;
+			get_multi_xcmp_funds(alice.clone());
+
+			let action = Action::XCMP {
+				destination,
+				schedule_fee: NATIVE_LOCATION,
+				execution_fee: AssetPayment { asset_location: destination.into(), amount: 10 },
+				encoded_call: vec![3, 4, 5],
+				encoded_call_weight: Weight::from_parts(100_000, 0),
+				overall_weight: Weight::from_parts(200_000, 0),
+				schedule_as: None,
+				instruction_sequence: InstructionSequence::PayThroughSovereignAccount,
+			};
+
+			let result = <Test as crate::Config>::FeeHandler::pay_checked_fees_for(
+				&alice,
+				&action,
+				1,
+				|| {
+					spy += 1;
+					Ok("called")
+				},
+			);
+			assert_eq!(result.expect("success"), "called");
+			assert_eq!(spy, 1);
+		})
+	}
+
+	#[test]
+	fn call_pay_checked_fees_for_with_normal_flow_and_insufficent_execution_fee_will_fail() {
+		new_test_ext(0).execute_with(|| {
+			let destination = MultiLocation::new(1, X1(Parachain(PARA_ID)));
+			let alice = AccountId32::new(ALICE);
+			fund_account(&alice, 900_000_000, 1, Some(0));
+
+			let action = Action::XCMP {
+				destination,
+				schedule_fee: NATIVE_LOCATION,
```

Comment:
Let’s make sure there’s a case covering the scenarion, when schedule_fee is a foreign asset and its balance is insufficient, the code will also throw a InsufficientBalance error.

Reply:
Fixed